### PR TITLE
refactored Peierls to use log(stress) in the log strain rate function

### PR DESCRIPTION
**Pull request by bobmyhill**
Minor refactoring of the new `compute_exact_log_strain_rate_and_derivative()` function in Peierls creep to use the log stress. This has two benefits:
- The function now returns F(x) and F'(x), rather than F(exp(x)) and F'(exp(x)). This will make it easier to use SUNDIALS in the future.
- It requires slightly fewer operations. 

FYI @lhy11009 

* [x] I have read the guidelines in our [CONTRIBUTING.md](../blob/main/CONTRIBUTING.md) document.
* [x] I have followed the [instructions for indenting my code](../blob/main/CONTRIBUTING.md#making-aspect-better).
* [x] I have tested my new feature locally to ensure it is correct.

### CODE REVIEW

File: source/material_model/rheology/peierls_creep.cc [hunk 1]
```diff
@@ -187,11 +187,12 @@ namespace aspect
         // Apply a strict cutoff if this option is chosen by user. A strain rate cutoff
         // will be first computed and then compared to the input strain rate. A cutoff
         // on stress will be triggered if the input strain rate is smaller.
+        double log_strain_rate = std::log(strain_rate);
```
Comment (bangerth) [inline source/material_model/rheology/peierls_creep.cc:190]:
```suggestion
        const double log_strain_rate = std::log(strain_rate);
```

**Review by bangerth — APPROVED** [review]

**Comment by lhy11009** [comment]
@bobmyhill Thanks for notifying me, I could see your point very clearly in the PR.

**Comment by gassmoeller** [comment]
This looks ready, seems like we just forgot to merge it after the hackathon.

**Pull request merged**